RunningOrder/Sprint/Managers/SprintManager.swift

```diff
@@ -34,13 +40,35 @@ final class SprintManager: ObservableObject {
         return saveSprintPublisher.eraseToAnyPublisher()
     }
 
-    func loadData(from space: Space) {
-        return service
-            .fetchAll(from: space.id)
-            .replaceError(with: [])
-            .receive(on: DispatchQueue.main)
-            .assign(to: \.sprints, onStrong: self)
-            .store(in: &cancellables)
+    func updateData(with updatedRecords: [CKRecord]) {
+        for updatedRecord in updatedRecords {
+            do {
+                let sprint = try Sprint(from: updatedRecord)
+                if let index = sprints.firstIndex(where: { $0.id == sprint.id }) {
+                    DispatchQueue.main.async {
+                        self.sprints[index] = sprint
+                    }
+                } else {
+                    Logger.verbose.log("sprint with id \(sprint.id) not found, so appending it to existing sprint list")
+                    DispatchQueue.main.async {
+                        self.sprints.append(sprint)
+                    }
+                }
+            } catch {
+                Logger.error.log(error)
```

Comment:
would we want proper error handling?

Reply:
in this particular case, I think this is fine to just log the error